src/lib/badge/badge.scss

```diff
@@ -0,0 +1,64 @@
+$mat-badge-size: 22px !default;
+
+.mat-badge {
+  position: relative;
+  display: inline-block;
+
+  &.mat-badge-above {
+    .mat-badge-content {
+      top: -$mat-badge-size / 2;
+    }
+  }
+  
+  &.mat-badge-below {
+    .mat-badge-content {
+      bottom: -$mat-badge-size / 2;
+    }
+  }
+
+  &.mat-badge-before {
+    margin-left: $mat-badge-size;
+    .mat-badge-content {
+      left: -$mat-badge-size;
+    }
+  }
+
+  &.mat-badge-after {
+    margin-right: $mat-badge-size;
+    .mat-badge-content {
+      right: -$mat-badge-size;
+    }
+  }
+
+  &.mat-badge-overlap {
+    &.mat-badge-before {
+      margin-left: $mat-badge-size / 2;
+      .mat-badge-content {
+        left: -$mat-badge-size / 2;
+      }
+    }
+  
+    &.mat-badge-after {
+      margin-right: $mat-badge-size / 2;
+      .mat-badge-content {
+        right: -$mat-badge-size / 2;
+      }
+    }
+  }
+
+  .mat-badge-content {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: center;
```

Comment:
Does this really need flexbox? It can be done simpler with `text-align` and `line-height`.

Reply:
It now supports font icons and svgs, so its needed now.